pages/learn/miscellaneous-examples.md

```diff
@@ -157,3 +174,243 @@ We also introduce the following with this example:
+
+```json
+{
+  "$id": "https://example.com/regex-pattern.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Regular Expression Pattern",
+  "type": "object",
+  "properties": {
+    "code": {
+      "type": "string",
+      "pattern": "^[A-Z]{3}-\\d{3}$"
+    }
+  }
+}
+```
+
+**Data**
+
+```json
+{
+  "code": "ABC-123"
+}
+```
+
+The provided data, "ABC-123", satisfies this pattern defined in the schema.
+
+
+## Complex object with nested properties
+
+The schema below represents a complex object with various properties including `name`, `age`, `address`, and `hobbies`. The `address` property is an object with nested properties, and the `hobbies` property is an array of strings. The `name` and `age` properties are required.
+
+```json
+{
+  "$id": "https://example.com/complex-object.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Complex Object",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "age": {
+      "type": "integer",
+      "minimum": 0
+    },
+    "address": {
+      "type": "object",
+      "properties": {
+        "street": {
+          "type": "string"
+        },
+        "city": {
+          "type": "string"
+        },
+        "state": {
+          "type": "string"
+        },
+        "postalCode": {
+          "type": "string",
+          "pattern": "\\d{5}"
+        }
+      },
+      "required": ["street", "city", "state", "postalCode"]
+    },
+    "hobbies": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": ["name", "age"]
+}
+```
+
+**Data**
+
+```json
+{
+  "name": "John Doe",
+  "age": 25,
+  "address": {
+    "street": "123 Main St",
+    "city": "New York",
+    "state": "NY",
+    "postalCode": "10001"
+  },
+  "hobbies": ["reading", "running"]
+}
+```
+
+The provided data conforms to the schema by including values for the required properties and ensuring the `age` is an integer greater than or equal to zero. The `address` object contains all the necessary properties, and the `hobbies` property is an array of strings.
+
+
+## Conditional validation with dependencies
+
+The schema below defines an object with two properties: `isMember` and `membershipNumber`. The `membershipNumber` property is required and must have a length of 10 characters only when the `isMember` property is set to true.
+
+```json
+{
+  "$id": "https://example.com/conditional-validation.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Conditional Validation",
+  "type": "object",
+  "properties": {
+    "isMember": {
+      "type": "boolean"
+    },
+    "membershipNumber": {
+      "type": "string",
+      "maxLength": 10,
+      "minLength": 10,
+      "dependencies": {
+        "isMember": true
```

Comment:
> ... only when the `isMember` property is set to true.

This isn't a correct usage of `dependencies`.  `dependencies doesn't support what you're claiming.  You'd need the proposed `propertyDependencies` keyword to match on property values like this.

`dependentProperties` contains lists of properties that are required when another property is present.

```json
{
  "dependentProperties": {
    "foo": [ "bar" ]
  }
}
```

This requires that `bar` be present when `foo` is present, but it doesn't require that `foo` or `bar` have any particular value, just that they exist.  Think of it as a conditional `required`.

`dependentSchemas` contains a subschema that the instance must also match if a property is present.

```json
{
  "dependentSchemas": {
    "foo": { "minProperties": 7 }
  }
}
```

This requires that the instance have 7 properties if `foo` is present.

Reply:
@aku1310 I mistyped on this at first and put `dependentProperties` when I meant `dependentRequired`.  I've corrected the above but wanted to explicitly say something in case you've already started your changes.